internal/venice-common/src/main/java/com/linkedin/venice/controller/converters/HttpConvertersRegistry.java

```diff
@@ -0,0 +1,42 @@
+package com.linkedin.venice.controller.converters;
+
+import com.linkedin.venice.controller.requests.ControllerHttpRequest;
+import com.linkedin.venice.controller.requests.ControllerRequest;
+import com.linkedin.venice.controller.requests.CreateStoreRequest;
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * A registry to hold all the converters for {@link ControllerRequest} to Http specific request.
+ * New addition of converters need to be registered below under the
+ * {@link #initialize()}. e.g.,
+ * <pre>
+ *   // Register sample request converter
+ *   registerRequestConverter(NewApiRequest.class, ConverterUtil::convertNewApiToHttpRequest)
+ * </pre>
+ */
+public class HttpConvertersRegistry {
+  private final Map<Class<?>, RequestConverter<? extends ControllerRequest, ControllerHttpRequest>> requestRegistry =
+      new HashMap<>();
```

Comment:
The `GrpcConvertersRegistry` has both request and response converters. Does this one likewise need to have both?

Reply:
I guess I just don't understand why we need it for one but not the other...?

Reply:
that is because, we decided to use a separate response data model for gRPC and `ControllerResponse` POJO isn't suitable, unless, we loose the typing information on the gRPC path and have responses represent byte[] in which case we can reuse all of it; 

We decided as a team to follow gRPC convention and make the new request, response strongly typed.